docs/data/date-pickers/custom-field/behavior-button/MaterialDateRangePicker.tsx

```diff
@@ -0,0 +1,86 @@
+import * as React from 'react';
+import { Dayjs } from 'dayjs';
+import Button from '@mui/material/Button';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import {
+  DateRangePicker,
+  DateRangePickerProps,
+  DateRangePickerFieldProps,
+} from '@mui/x-date-pickers-pro/DateRangePicker';
+import { useValidation } from '@mui/x-date-pickers/validation';
+import { validateDateRange } from '@mui/x-date-pickers-pro/validation';
+import {
+  useSplitFieldProps,
+  useParsedFormat,
+  usePickersContext,
+} from '@mui/x-date-pickers/hooks';
+
+function ButtonDateRangeField(props: DateRangePickerFieldProps<Dayjs>) {
+  const { internalProps, forwardedProps } = useSplitFieldProps(props, 'date');
+  const { value, timezone, format } = internalProps;
+  const {
+    InputProps,
+    slotProps,
+    slots,
+    ownerState,
+    label,
+    focused,
+    name,
+    ...other
+  } = forwardedProps;
+
+  const pickersContext = usePickersContext();
+
+  const parsedFormat = useParsedFormat(internalProps);
+  const { hasValidationError } = useValidation({
+    validator: validateDateRange,
+    value,
+    timezone,
+    props: internalProps,
+  });
+
+  const handleTogglePicker = (event: React.UIEvent) => {
+    if (pickersContext.open) {
+      pickersContext.onClose(event);
+    } else {
+      pickersContext.onOpen(event);
+    }
+  };
+
+  const valueStr = (value ?? [null, null])
+    .map((date) => (date == null ? parsedFormat : date.format(format)))
+    .join(' – ');
+
+  return (
+    <Button
+      {...other}
+      variant="outlined"
+      color={hasValidationError ? 'error' : 'primary'}
+      ref={InputProps?.ref}
+      onClick={handleTogglePicker}
+    >
+      {label ? `${label}: ${valueStr}` : valueStr}
+    </Button>
+  );
+}
+
+// Will be removed before the end of the alpha since single input will become the default fields.
```

Comment:
WDYT, is it worth adding a `TODO v8` prefix?
```suggestion
// TODO v8: Will be removed before the end of the alpha since single input will become the default field.
```